protocol/synthetix/test/integration/modules/core/RewardsManagerModule.test.ts

```diff
@@ -668,6 +670,39 @@ describe('RewardsManagerModule', function () {
             ethers.utils.parseEther('1')
           )
       ).wait();
+
+      // have a new user join in after the distribution to verify
+      // they dont get unjustified stake
+      await (
+        await systems()
+          .CollateralMock.connect(owner)
+          .mint(await owner.getAddress(), rewardAmount)
+      ).wait();
+      await (
+        await systems()
+          .CollateralMock.connect(owner)
+          .approve(systems().Core.address, ethers.constants.MaxUint256)
+      ).wait();
+      await (
+        await systems().Core.connect(owner).deposit(2222, collateralAddress(), rewardAmount)
+      ).wait();
+      await (
+        await systems()
+          .Core.connect(owner)
+          .delegateCollateral(
+            2222,
+            poolId,
+            collateralAddress(),
+            rewardAmount,
+            ethers.utils.parseEther('1')
+          )
+      ).wait();
+
+      const availableRewards = await systems()
+        .Core.connect(user1)
+        .getAvailableRewards(2222, poolId, collateralAddress(), RewardDistributorPoolLevel.address);
+
+      console.log('POST AVAILABLE REWARDED', availableRewards);
```

Comment:
Asert?

Reply:
sorry no that was just for sanity debug. I removed the console log and the getavailablerewards call